template/extra_runtime.go

```diff
@@ -377,14 +377,20 @@ func (t *Template) tryConvert(value string) interface{} {
 }
 
 // Execute the command (command could be a file, a template or a script) and convert its result as data if possible
-func (t *Template) exec(command string, args ...interface{}) (result interface{}, err error) {
-	if result, err = t.run(command, args...); err == nil {
-		if result == nil {
-			return
-		}
-		result = t.tryConvert(result.(string))
+func (t *Template) exec(command string, args ...interface{}) (interface{}, error) {
+	commandOutput, err := t.run(command, args...)
+	if err != nil || commandOutput == nil {
```

Comment:
I checked the conditions and they changed just a weee bit in a way that I don't see problematic, but I just wanted to know if it was intentional:

We went from `err == nil and result == nil` to `err != nil or result == nil`
From what I can see, we only win from this since we won't convert `nil` command outputs, but I have a limited understanding and prefer to nitpick 😉.

Reply:
You're right, that's a mighty subtle difference but it might have some effect. There's 4 options here so I think that it's worth just going through the truth table to make sure

commandOuput | err | what happens?
--- | --- | ---
`nil` | `nil` | We return `nil, nil` without parsing
`nil` | not `nil` | We return `nil, err` without parsing
not `nil` | `nil` | We attempt to parse `commandOutput`
not `nil` | not `nil` | We return `commandOutput, nil` without parsing

How does that compare to the original?

`result` | `err` | What happens?
--- | --- | ---
`nil` | `nil` | We return `nil, nil` without parsing
`nil` | not `nil` | We return `nil, err` without parsing
not `nil` | `nil` | We attempt to parse `result`
not `nil` | not `nil` | We return `result, err` without parsing

Looks like they have the same behaviour.